.gitlab-ci.yml

```diff
@@ -178,16 +174,16 @@ opam-boot:
   cache:
     paths: &cache-paths
       - .opamcache
-    key: main
+    key: main-bionic-V2018-04-28-A
```

Comment:
I don't like this system. Updating the keys everytime we need to is going to be a pain.
How about instead of using `key` we do
- at the beginning of opam-boot script, `if [ "$(cat .opamcache/key)" != "$(md5sum .gitlab-ci.yml)" ]; then rm -rf .opamcache; fi`
- at the end `md5sum .gitlab-ci.yml > .opamcache/key`

Reply:
Not so much of pain IMO but indeed prone to manual errors. I dunno, we can do the hack too if you prefer until `key` gets stronger in GitLab. Anyways #7383 should help with some problems IMHO.

Reply:
If you go with the yml key you should use a variable `CACHEKEY: "V..."` instead of repeating it 3 times.
Also I don't see the point of putting "bionic" in there.

Reply:
Good idea, key variable refactored.

> Also I don't see the point of putting "bionic" in there.

I am trying to put as much info as I can; bionic is useful as for example having a different key let's say, for Nix.